libs/auth/test/utils.ts

```diff
@@ -10,17 +12,47 @@ export function numericArray(input: {
   return Array.from<number>({ length: input.length }).fill(input.value ?? 0);
 }
 
+/**
+ * The card API with all methods mocked using our custom libs/test-utils mocks
+ */
+export interface MockCard {
+  getCardStatus: MockFunction<Card['getCardStatus']>;
+  checkPin: MockFunction<Card['checkPin']>;
+  program: MockFunction<Card['program']>;
+  readData: MockFunction<Card['readData']>;
+  writeData: MockFunction<Card['writeData']>;
+  clearData: MockFunction<Card['clearData']>;
+  unprogram: MockFunction<Card['unprogram']>;
+}
+
 /**
  * Builds a mock card instance
  */
-export function buildMockCard(): Card {
+export function buildMockCard(): MockCard {
   return {
-    getCardStatus: jest.fn(() => Promise.resolve({ status: 'no_card' })),
-    checkPin: jest.fn(),
-    program: jest.fn(),
-    readData: jest.fn(),
-    writeData: jest.fn(),
-    clearData: jest.fn(),
-    unprogram: jest.fn(),
+    getCardStatus: mockFunction<Card['getCardStatus']>('getCardStatus'),
+    checkPin: mockFunction<Card['checkPin']>('checkPin'),
+    program: mockFunction<Card['program']>('program'),
+    readData: mockFunction<Card['readData']>('readData'),
+    writeData: mockFunction<Card['writeData']>('writeData'),
+    clearData: mockFunction<Card['clearData']>('clearData'),
+    unprogram: mockFunction<Card['unprogram']>('unprogram'),
   };
 }
+
+/**
+ * Asserts that all the expected calls to all the methods of a mock card were made
+ */
+export function mockCardAssertComplete(mockCard: MockCard): void {
+  for (const method of [
+    'getCardStatus',
+    'checkPin',
+    'program',
+    'readData',
+    'writeData',
+    'clearData',
+    'unprogram',
+  ] as const) {
```

Comment:
Any way to make sure this doesn't get out of date if we add a method?

Reply:
Good call! Came up with this: https://github.com/votingworks/vxsuite/pull/3145/commits/20cc3e4648e4404eb05344c3f31344986e9d6e3e

Reply:
^ (TS will complain if we're not exhaustive)

<img width="965" alt="example" src="https://user-images.githubusercontent.com/12616928/225718900-ee984d69-0272-4153-b5b4-5628c68ad861.png">

Reply:
Nice, that's clever.

I looked to see how we do it in the Grout client and [it's a bit simpler](https://github.com/votingworks/vxsuite/blob/7ca8e5abd1953b51f8daf71e04be4e7e8b8385ed/libs/grout/test-utils/src/mock_client.ts#L70-L74). Wondering if the same approach would work here

Reply:
Ah nice, yeah much simpler: https://github.com/votingworks/vxsuite/pull/3145/commits/94ff53443e3197b6fb30700c0beff85b82e1adc0